src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -942,25 +944,188 @@ private void DecodeModuleInitializerAttribute(DecodeWellKnownAttributeArguments<
 
         private void DecodeInterceptsLocationAttribute(DecodeWellKnownAttributeArguments<AttributeSyntax, CSharpAttributeData, AttributeLocation> arguments)
         {
-            Debug.Assert(arguments.AttributeSyntaxOpt is object);
             Debug.Assert(!arguments.Attribute.HasErrors);
-            var attributeData = arguments.Attribute;
-            var attributeArguments = attributeData.CommonConstructorArguments;
-            if (attributeArguments is not [
+            var constructorArguments = arguments.Attribute.CommonConstructorArguments;
+            if (constructorArguments is [
                 { Type.SpecialType: SpecialType.System_String },
                 { Kind: not TypedConstantKind.Array, Value: int lineNumberOneBased },
                 { Kind: not TypedConstantKind.Array, Value: int characterNumberOneBased }])
             {
-                // Since the attribute does not have errors (asserted above), it should be guaranteed that we have the above arguments.
-                throw ExceptionUtilities.Unreachable();
+                DecodeInterceptsLocationAttributeExperimentalCompat(arguments, attributeFilePath: (string?)constructorArguments[0].Value, lineNumberOneBased, characterNumberOneBased);
+            }
+            else
+            {
+                Debug.Assert(arguments.Attribute.AttributeConstructor.Parameters is [{ Type.SpecialType: SpecialType.System_Int32 }, { Type.SpecialType: SpecialType.System_String }]);
+                DecodeInterceptsLocationChecksumBased(arguments, version: (int)constructorArguments[0].Value!, data: (string?)constructorArguments[1].Value);
             }
+        }
 
+        private void DecodeInterceptsLocationChecksumBased(DecodeWellKnownAttributeArguments<AttributeSyntax, CSharpAttributeData, AttributeLocation> arguments, int version, string? data)
+        {
+            var diagnostics = (BindingDiagnosticBag)arguments.Diagnostics;
+            var attributeNameSyntax = arguments.AttributeSyntaxOpt!.Name; // used for reporting diagnostics
+            var attributeLocation = attributeNameSyntax.Location;
+
+            if (version != 1)
+            {
+                diagnostics.Add(ErrorCode.ERR_InterceptsLocationUnsupportedVersion, attributeLocation, version);
+                return;
+            }
+
+            if (InterceptableLocation1.Decode(data, attributeLocation, diagnostics) is not var (hash, position, displayFileName))
```

Comment:
Consider reporting the diagnostic just once here (`diagnostics.Add(ErrorCode.ERR_InterceptsLocationDataInvalidFormat, diagnosticLocation);`) instead of inside `Decode` to reduce repetition.